components/support/utils/src/main/java/mozilla/components/support/utils/URLStringUtils.kt

```diff
@@ -28,4 +29,12 @@ object URLStringUtils {
         }
         return uri.toString()
     }
+
+    private val isURLLenient by lazy {
+        // Be lenient about what is classified as potentially a URL. Anything that contains a :,
+        // ://, or . and has no internal spaces is potentially a URL. Use java.util.regex because
```

Comment:
"Use java.util.regex over Kotlin Regex" (iiuc)

To be fair, Kotlin Regex probably uses java.util.Regex under the hood and our unit tests would catch if that changes.